plasma_framework/contracts/src/framework/BlockController.sol

```diff
@@ -63,8 +78,8 @@ contract BlockController is Operated, VaultRegistry {
      * @dev see discussion in https://github.com/omisego/plasma-contracts/issues/233
      * @param _blockRoot Merkle root of the plasma block.
      */
-    function submitBlock(bytes32 _blockRoot) external {
-        require(msg.sender == authority, "Can be called only by the Authority.");
+    function submitBlock(bytes32 _blockRoot) external onlyFrom(authority) {
+        require(isChildChainActivated == true, "Child chain has not been activate by authority address yet");
```

Comment:
Do we need this check? I am ok if we make this implicit and enforce in child chain

Reply:
I guess it's safer ... :)

Reply:
I added this check in both `submitBlock` and `submitDepositBlock`. So is your proposal to enforce this implicit on child chain for `submitBlock` but keep the `submitDepositBlock`? And we are doing this to save some gas per block right?

Reply:
Yeah I was thinking about gas ... but safer is always good.